linera-execution/solidity/Linera.sol

```diff
@@ -7,123 +7,177 @@ import "./LineraTypes.sol";
+        LineraTypes.ContractRuntimePrecompile memory contract_ = LineraTypes.ContractRuntimePrecompile_case_send_message(send_message_);
+        LineraTypes.RuntimePrecompile memory input1 = LineraTypes.RuntimePrecompile_case_contract(contract_);
+        bytes memory input2 = LineraTypes.bcs_serialize_RuntimePrecompile(input1);
         (bool success, bytes memory output) = precompile.call(input2);
         require(success);
         require(output.length == 0);
     }
 
     function message_id() internal returns (LineraTypes.opt_MessageId memory) {
         address precompile = address(0x0b);
-        bytes memory input1 = new bytes(2);
-        input1[0] = bytes1(uint8(1));
-        input1[1] = bytes1(uint8(3));
-        (bool success, bytes memory output1) = precompile.call(input1);
+        LineraTypes.ContractRuntimePrecompile memory contract_ = LineraTypes.ContractRuntimePrecompile_case_message_id();
+        LineraTypes.RuntimePrecompile memory input1 = LineraTypes.RuntimePrecompile_case_contract(contract_);
+        bytes memory input2 = LineraTypes.bcs_serialize_RuntimePrecompile(input1);
+        (bool success, bytes memory output) = precompile.call(input2);
         require(success);
-        LineraTypes.opt_MessageId memory output2 = LineraTypes.bcs_deserialize_opt_MessageId(output1);
-        return output2;
+        return LineraTypes.bcs_deserialize_opt_MessageId(output);
     }
 
     function message_is_bouncing() internal returns (LineraTypes.MessageIsBouncing memory) {
         address precompile = address(0x0b);
-        bytes memory input1 = new bytes(2);
-        input1[0] = bytes1(uint8(1));
-        input1[1] = bytes1(uint8(4));
-        (bool success, bytes memory output1) = precompile.call(input1);
+        LineraTypes.ContractRuntimePrecompile memory contract_ = LineraTypes.ContractRuntimePrecompile_case_message_is_bouncing();
+        LineraTypes.RuntimePrecompile memory input1 = LineraTypes.RuntimePrecompile_case_contract(contract_);
+        bytes memory input2 = LineraTypes.bcs_serialize_RuntimePrecompile(input1);
+        (bool success, bytes memory output) = precompile.call(input2);
+        require(success);
+        return LineraTypes.bcs_deserialize_MessageIsBouncing(output);
+    }
+
+    function linera_emit(bytes memory stream_name, bytes memory value) internal returns (uint32) {
+        address precompile = address(0x0b);
+        LineraTypes.StreamName memory stream_name2 = LineraTypes.StreamName(stream_name);
+        LineraTypes.ContractRuntimePrecompile_Emit memory emit_ = LineraTypes.ContractRuntimePrecompile_Emit(stream_name2, value);
+        LineraTypes.ContractRuntimePrecompile memory contract_ = LineraTypes.ContractRuntimePrecompile_case_emit(emit_);
+        LineraTypes.RuntimePrecompile memory input1 = LineraTypes.RuntimePrecompile_case_contract(contract_);
```

Comment:
Good these are hidden behind a better API b/c I'd have no idea what's going on in here.

Reply:
Yes, that handwritten code was looking very error-prone.
Better to use some computer generated code.